erpnext/accounts/report/gross_profit/gross_profit.py

```diff
@@ -748,12 +748,13 @@ def get_last_purchase_rate(self, item_code, row):
 			.inner_join(purchase_invoice)
 			.on(purchase_invoice.name == purchase_invoice_item.parent)
 			.select(
-				purchase_invoice.name,
 				purchase_invoice_item.base_rate / purchase_invoice_item.conversion_factor,
 			)
 			.where(purchase_invoice.docstatus == 1)
 			.where(purchase_invoice.posting_date <= self.filters.to_date)
 			.where(purchase_invoice_item.item_code == item_code)
+			.where(purchase_invoice.is_return == 0)
+			.where(purchase_invoice_item.parenttype == "Purchase Invoice")
```

Comment:
Is this necessary? I don't think `Purchase` Invoice Item` is used anywhere else

Reply:
I had an issue where user was using Item Table in custom doctype.